app/src/main/kotlin/com/wire/android/ui/common/topappbar/AppTopBarWithSearchBarLayout.kt

```diff
@@ -0,0 +1,269 @@
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.animation.animateContentSize
+import androidx.compose.animation.core.snap
+import androidx.compose.animation.expandVertically
+import androidx.compose.animation.shrinkVertically
+import androidx.compose.foundation.background
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.collectIsPressedAsState
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.LocalTextStyle
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import com.wire.android.R
+import com.wire.android.ui.common.SearchBarInput
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.theme.wireColorScheme
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.scan
+
+
+/**
+ * AppTopBarWithSearchBarLayout is a "wrapper" around the [content] that should have a TopBar with a SearchBar
+ * on the Top of the [content]. To collapse the searchbar when the user scrolls down on the list within the [content]
+ * it is necessary to pass a [scrollPosition] usually it is a firstVisibleItemIndex coming out of the list within the content.
+ * AppTopBarWithSearchBarLayout also exposes [searchQuery] as well as [onSearchQueryChanged] so that we are able to manipulate
+ * the [searchQuery] outside of the AppTopBarWithSearchBarLayout for example a ViewModel. Beside collapsing the searchbar when scrolling
+ * through the list, AppTopBarWithSearchBarLayout also hides/shows the TopBar whenever the SearchBarInput is clicked so that we
+ * can transit to the state where we provide the search query to the searchbar.
+ */
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun AppTopBarWithSearchBarLayout(
```

Comment:
Maybe for more semantic coherence with other composables rename to:
```suggestion
fun AppTopBarWithSearchBar(
```